lib/kafo/data_type_parser.rb

```diff
@@ -8,9 +8,26 @@ class DataTypeParser
 
     def initialize(manifest)
       @logger = KafoConfigure.logger
-      @types = {}
+
+      lines = []
+      last_line = ''
```

Comment:
Using `'' << 'x'` fails with frozen strings so I've learned to avoid that and use `+=` or explicitly mark the string as mutable. Right now the file doesn't have the `frozen_string_literal: true` header, but if it's added, things will break. The easiest is to mark it as mutable:

```suggestion
      last_line = +''
```

Reply:
done